cores/arduino/Uart.cpp

```diff
@@ -64,6 +84,13 @@ void Uart::IrqHandler()
 {
   if (sercom->availableDataUART()) {
     rxBuffer.store_char(sercom->readDataUART());
+
+    if (uc_pinRTS != NO_RTS_PIN) {
+      // if there is NOT enough space in the RX buffer, de-assert RTS
+      if (rxBuffer.availableForStore() < RTS_RX_THRESHOLD) {
+        digitalWrite(uc_pinRTS, HIGH);
```

Comment:
`digitalWrite` may be slow, we should explore direct register access.

Reply:
I've pushed a commit to leverage SERCOM h/w capabilities.